Add steps for local data ingestion in docs

sumairq opened the pull request:
## This PR updates the docs on steps to ingest data locally.

It is a solution for https://bugzilla.mozilla.org/show_bug.cgi?id=1850748

- It includes steps to help contributors set up a local data ingestion environment.

EvaBardou committed:
Restore the switch to OpenAPI (require missing dependency)

Commit:
Serve ReDoc inside react app

beatrice-acasandrei committed:
Add application field

beatrice-acasandrei committed:
Shorten field name

Archaeopteryx committed:
Upgrade CircleCI remote docker from 19.03.14 to 20.10.23

sumairq committed:
Add steps for local data ingestion in docs

gmierz commented on docs/pulseload.md:
```diff
@@ -62,3 +62,60 @@ ex: <https://community-tc.services.mozilla.com/pulse-messages/>
 [pulse guardian]: https://pulseguardian.mozilla.org/whats_pulse
 [yml schema]: https://github.com/mozilla/treeherder/blob/master/schemas/pulse-job.yml
+
+**Here is an example flow you can use:**
```
We can remove this line

gmierz commented on docs/pulseload.md:
```diff
@@ -62,3 +62,60 @@ ex: <https://community-tc.services.mozilla.com/pulse-messages/>
+
+**Here is an example flow you can use:**
+
+## Setup
```
Change this to `## Setup for Perfherder Development`

gmierz commented on docs/pulseload.md:
```diff
@@ -62,3 +62,60 @@ ex: <https://community-tc.services.mozilla.com/pulse-messages/>
+
+## Setup
+
+(These steps should help you set up everything you need to ingest data locally while working on perfherder)
```
We could remove the parentheses for this line

gmierz commented on docs/pulseload.md:
```diff
@@ -62,3 +62,60 @@ ex: <https://community-tc.services.mozilla.com/pulse-messages/>
+
+**Run the db viewer**
+
+You can use **dbeaver-ce**
```
I recommend only mentioning that you can use any database viewer of your choice here. You could mention `dbeaver-ce` in parentheses like `(e.g. dbeaver-ce, mysql workbench, etc.)`

gmierz commented on docs/pulseload.md:
```diff
@@ -62,3 +62,60 @@ ex: <https://community-tc.services.mozilla.com/pulse-messages/>
+
+---
+
+##### Run the following in separate window while running above to do ingestion
```
Below this, we should add a comment saying that these commands actually perform the fetches for the data, and that they are run sequentially in the same window. The first command makes the second one run faster. Also, we should mention here the `--enable-eager-celery` triggers the log parsing which is required to capture the `PERFHERDER_DATA` output.

gmierz reviewed (changes requested):
Nice work @sumairq :) just a few additional things we'll need to add to make it clearer for others.

jmaher committed:
Expand inter-MITTEN-t logic to account for confirm-failures. (#7773)

Archaeopteryx commented on docs/pulseload.md:
````diff
@@ -62,3 +62,60 @@ ex: <https://community-tc.services.mozilla.com/pulse-messages/>
+docker-compose exec backend ./manage.py ingest push -p autoland -r 1ee42a54a431acdd6cbe43b49de0237fe67eddd9 -a --enable-eager-celery
+```
+
+**For ingesting multple pushes:**
````
Typo.

Archaeopteryx commented on docs/pulseload.md:
````diff
@@ -62,3 +62,60 @@ ex: <https://community-tc.services.mozilla.com/pulse-messages/>
+
+**Ingest all the tasks, and run celery to trigger the log parsing, and performance data ingestion:**
+
+```bash
````
There should be a warning this can take a long time to ingest and parse everything.

Archaeopteryx commented on docs/pulseload.md:
````diff
@@ -62,3 +62,60 @@ ex: <https://community-tc.services.mozilla.com/pulse-messages/>
+**Run each of these commands in multiple windows:**
+
+```bash
+docker-compose up --build
````
`--build` has been already used above and should not be needed here.

Archaeopteryx commented on docs/pulseload.md:
```diff
@@ -19,7 +19,7 @@ export PROJECTS_TO_INGEST=autoland,try
 username and password. Remember these as you'll use them in the next step.
 This is recommended, because using the default value **MAY** cause you to miss some data,
-if it was already ingested by another user  Unfortunately, **Pulse** doesn't support creating
+if it was already ingested by another user Unfortunately, **Pulse** doesn't support creating
```
This needs a full stop between the two sentences.

Archaeopteryx reviewed (commented):
Thank you for extending the documentation.

sumairq committed:
Address comments

gmierz commented:
@sumairq looking at the other documentation we have, and I'm wondering if it would make more sense to have this information on this page (maybe in the ingesting pushes section): https://treeherder.readthedocs.io/installation.html

What do you think?

vrigal committed:
Upgrade eslint

vrigal committed:
Upgrade @mozilla/glean

vrigal committed:
Upgrade webpack-cli

vrigal committed:
Upgrade webpack

Archaeopteryx reviewed (approved):
Thank you.

EvaBardou committed:
Serve Redoc in a dedicated chunk

renovate[bot] committed:
Update rabbitmq Docker tag to v3.11.22

renovate[bot] committed:
Update redis Docker tag to v7.0.12

renovate[bot] committed:
Update dependency mobx to v6.10.2

renovate[bot] committed:
Update redis Docker tag to v7.0.13

alexandru-io committed:
Add tests for the Debug Tools column in alerts view (#7731)

Bug 1841022 - Add tests for the Debug Tools column in alerts view

renovate[bot] committed:
Update dependency mobx to v6.10.2 (#7800)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Pin dependency styled-components to 5.3.11 (#7799)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @mozilla/glean to v2.0.2

renovate[bot] committed:
Update dependency url to v0.11.3

renovate[bot] committed:
Update rabbitmq Docker tag to v3.11.23

CanadaHonk committed:
Fix select all test checkboxes not scrolling properly

Previously, select all checkboxes in the tests tab would always stay on the same position on screen due to a missing positive relative.

CanadaHonk committed:
Potential fix for Bug 1827920 (#7809)

Bug 1827920 - also get the `updatePinnedJob` method in the local scope to fix sporadic push health issue

The import was being used instead of the prop tied to the store resulting in the `updatePinnedJob` prop sometimes having the value of `null` when it is attempted to be called later inside FailureSummaryTab.

Archaeopteryx committed:
Renovate node20 (#7813)

* Update Node.js to v20.6.1

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

alexandru-io committed:
Add TaskclusterMetadata to the alert summary API endpoint (#7740)

Bug 1838836 - Add TaskclusterMetadata to the alert summary API endpoint

vrigal committed:
Optimize summarygroup endpoint (#7812)

* Base the query on JobLog table

* Filter from the database directly

* Refactor the Python code

* Add distinct clause

* Force ordering of the entire payload

* Add a comment for summarygroup result direct annotation

jmaher committed:
Fix mitten to work with -cf in the job names (#7810)

* Bug 1850785 - Allow for Marionette failures to be filed as single tracking bugs. r=aryx

* allow mitten to work with cf jobs, fix bug with -cf at end of jobname.

renovate[bot] committed:
Update dependency docs/mkdocs to v1.5.3

gmierz commented:
> a link to the detailed setup in the `Loading pulse data` section?

I'm not sure about this, why do you think the `Loading pulse data` section would be better? While navigating locally, I found that it's a bit harder to find it here versus the installation page.

sumairq committed:
Add steps to installation section

gmierz commented:
@sumairq would you be able to fix up this PR to only show the files modified by your PR? It looks like there are merge conflicts as well that need to be resolved. I think a forced push from an up-to-date branch could help.

beatrice-acasandrei commented:
Due to the recent inactivity we can close this PR, this issue will be wrapped up in https://github.com/mozilla/treeherder/pull/9477.
